homeassistant/components/devolo_home_control/devolo_device.py

```diff
@@ -0,0 +1,84 @@
+"""Base class for a device entity integrated in devolo Home Control."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .subscriber import Subscriber
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_BATTERY_LEVEL = "battery_level"
+
+
+class DevoloDeviceEntity(Entity):
+    """Representation of a sensor within devolo Home Control."""
+
+    def __init__(self, homecontrol, device_instance, element_uid, name, sync):
+        """Initialize a devolo device entity."""
+        self._device_instance = device_instance
+        self._name = name
+        self._unique_id = element_uid
+        self._homecontrol = homecontrol
+        self._available = device_instance.is_online()
+
+        # Get the brand and model information
+        self._brand = device_instance.brand
+        self._model = device_instance.name
+
+        self._state_attrs = {}
+        if hasattr(self._device_instance, "batteryLevel"):
+            self._state_attrs = {ATTR_BATTERY_LEVEL: self._device_instance.batteryLevel}
```

Comment:
Battery level should not be a state attribute but its own sensor entity.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Done in https://github.com/home-assistant/core/pull/36557